### PR TITLE
 Added various combinators, most of which are also present in Haskell

### PR DESCRIPTION
**Pull request by Synthetica9**

### CODE REVIEW

File: libs/contrib/Data/Combinators/Applicative.idr [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+||| for function composition:
+module Data.Combinators.Applicative
+
+import Data.Combinators
+
+%default total
+%access public export
+
+||| Pure is just const, also known as the K combinator
+pure : a -> b -> a
```
Comment (LeifW) [inline libs/contrib/Data/Combinators/Applicative.idr]: `pure : Applicative f => a -> f a` is already defined as a member of the `Applicative` typeclass in Prelude.

**Comment by eraserhd** [comment]
I think the Arrow ones are all already present in `Control.Arrow` in `base`.

**Comment by Synthetica9** [comment]
While you are right, these are only defined or `Morphism`, not over bare
`(->)`, because the compiler doesn't allow this.

**Comment by ahmadsalim** [comment]
Thanks for the PR.
Since these are in `contrib` I will just merge them.

Consider adding dependent versions as well.

**Pull request merged**